Extensions/SendTo/ActionViewController.swift

```diff
@@ -6,108 +6,38 @@ import UIKit
-    var clientPickerDelegate: ClientPickerViewControllerDelegate?
-    
-    var clients: [RemoteClient] = []
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        title = NSLocalizedString("Send To Device", tableName: "SendTo", comment: "Title of the dialog that allows you to send a tab to a different device")
-        refreshControl = UIRefreshControl()
-        refreshControl?.addTarget(self, action: "refresh", forControlEvents: UIControlEvents.ValueChanged)
-        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Cancel, target: self, action: "cancel")
-    }
-
-    override func viewWillAppear(animated: Bool) {
-        super.viewWillAppear(animated)
-        reloadClients()
-    }
-    
-    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
-        return 1
-    }
-    
-    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return clients.count
-    }
-    
-    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = UITableViewCell()
-        cell.textLabel?.font = UIFont(name: UIAccessibilityIsBoldTextEnabled() ? "HelveticaNeue-Medium" : "HelveticaNeue", size: 17)
-        cell.textLabel?.text = String(format: NSLocalizedString("Send to %@", tableName: "SendTo", comment: "Text in a table view row that lets the user pick a device to which to send a tab to"), clients[indexPath.row].name)
-        return cell
-    }
-    
-    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        tableView.deselectRowAtIndexPath(indexPath, animated: true)
-        clientPickerDelegate?.clientPickerViewController(self, didPickClients: [clients[indexPath.row]])
-    }
-    
-    private func reloadClients() {
-        profile?.getClients().upon({ result in
-            self.refreshControl?.endRefreshing()
-            if let c = result.successValue {
-                self.clients = c
-                dispatch_async(dispatch_get_main_queue()) {
-                    self.tableView.reloadData()
-                }
-            }
-        })
-    }
-    
-    func refresh() {
-        reloadClients()
-    }
-    
-    func cancel() {
-        self.extensionContext!.completeRequestReturningItems(nil, completionHandler: nil)
-    }
-}
-
-/*!
-The ActionViewController is the initial viewcontroller that is presented (full screen) when the share extension is activated.
-Depending on whether the user is logged in or not, this viewcontroller will present either a Login or ClientPicker.
-*/
+/// The ActionViewController is the initial viewcontroller that is presented (full screen) when the share extension
+/// is activated. Depending on whether the user is logged in or not, this viewcontroller will present either the
+/// InstructionsVC or the ClientPicker VC.
 
 @objc(ActionViewController)
```

Comment:
No this is a special case where ActionViewController is referenced from the Info.plist file. Without it, iOS cannot find the `NSExtensionPrincipalClass`. I'll double check if this is still the case now with Swift 1.2.